### DISCUSSION TITLE
Nested OR FIlter

### DISCUSSION THREAD

**Discussion by reharik** [prisma/prisma#10083] [Q&A]
can I do something like this?
```
{
	where: {
		AND: [
			{bla: 
				{equals: "blah"}
			},
			{bla2: 
				{equals: "blah"}
			},
			OR: [
				{bla3: 
					{equals: "blah"}
				},
				{bla4: 
					{equals: "blah"}
				},
			]
		]
	}
}
```
I'm trying to create the ability to do dynamic queries ( how I miss C# lamdas ).  and what I'me getting at is something like 

`if(bla === "blah" && blah2 === "blah" || bla3 === "blah"  || bla4 === "blah)`

or worse 

`if(bla === "blah" && blah2 === "blah" || (bla3 === "blah"  && bla4 === "blah))`

Perhaps this is asking too much. 
The thing is I may have a required "where" clause in a repository function such as organizationid === currentUser.organizationId.  And the client whats to add a filter such as FirstName === "blah".  So I don't know what kind of filter they would pass in.  I'm allowing for AND filters and OR filters to be passed in and if it was just those criteria it would be fine, but there are some of the "organizationId" type criteria occasionally too. 

Hope that makes sense and is not too abstract.
Thanks,

**Comment by austincrim** [comment]
Hi @reharik!

What you are describing is definitely possible with Prisma. You can deeply nest `AND`, `OR`, and `NOT` without an issue, as long as you can keep the boolean logic straight in your head 😄.

Let me know if you run into any trouble!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
